### PR TITLE
DT-23157 dischargewizard refactor

### PR DESCRIPTION
**Pull request by zacharymorel**
## Issue
https://github.com/department-of-veterans-affairs/va.gov-team/issues/23157

## Description
Refactoring Discharge code as I see it. 
Got rid of unused code, refactored classes to use functions using hooks.
Added classes on headers for screen reader friendiness. 

## Testing done
Ran unit, ran e2e tests, tested app locally.

## Screenshots
Screen Shot 2021-05-12 at 4 01 31 PM
Screen Shot 2021-05-12 at 4 01 47 PM
Screen Shot 2021-05-12 at 4 02 38 PM

## Acceptance criteria
- [x] App works as it does on prod

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by zacharymorel** [comment]
FYI, @department-of-veterans-affairs/vsa-public-websites-frontend there is a lot of files changed; however, the majority is adding a class to a header element.

File: src/applications/discharge-wizard/containers/FormPage.jsx [hunk 1]
```diff
@@ -1,37 +1,37 @@
...
-import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import React from 'react';
 
+// Relative Imports
 import { updateField } from '../actions';
 import FormQuestions from '../components/FormQuestions';
 
-export class FormPage extends Component {
-  componentDidMount() {
```
Comment (kelsonic) [inline src/applications/discharge-wizard/containers/FormPage.jsx:8]: Is this componentDidMount logic not needed anymore?

File: src/applications/discharge-wizard/containers/GuidancePage.jsx [hunk 2]
```diff
@@ -14,83 +15,71 @@ import StepTwo from '../components/gpSteps/StepTwo';
...
+  );
 
-  componentDidMount() {
-    localStorage.setItem('dw-viewed-guidance', true);
-    localStorage.setItem(
-      'dw-formValues',
-      JSON.stringify(this.props.formValues),
-    );
-
-    if (sessionStorage.getItem('dw-session-started')) {
```
Comment (kelsonic) [inline src/applications/discharge-wizard/containers/GuidancePage.jsx:31]: Same thing with these conditionals, not needed anymore?

**Review by kelsonic — APPROVED** [review]
lgtm, there's 2 comments with removed code that wasn't added back in so just want to confirm that was desired 🙂

File: src/applications/discharge-wizard/containers/GuidancePage.jsx (hunk 2 above)
Comment (zacharymorel) [reply src/applications/discharge-wizard/containers/GuidancePage.jsx:31]: Yep, it wasn't doing another that the localstorage was already doing!

File: src/applications/discharge-wizard/containers/FormPage.jsx (hunk 1 above)
Comment (zacharymorel) [reply src/applications/discharge-wizard/containers/FormPage.jsx:8]: Yes, same as the comment below!

**Pull request merged**